src/hooks/useGas.ts

```diff
@@ -60,21 +66,19 @@ export default function useGas({ nativeAsset }: { nativeAsset?: ParsedAddressAss
   const dispatch = useDispatch();
   const { nativeCurrency } = useAccountSettings();
 
-  // keep native assets up to date
-  useExternalToken({
-    address: BNB_MAINNET_ADDRESS,
-    chainId: ChainId.mainnet,
-    currency: nativeCurrency,
-  });
-  useExternalToken({
-    address: ETH_ADDRESS,
-    chainId: ChainId.mainnet,
-    currency: nativeCurrency,
-  });
-  useExternalToken({
-    address: MATIC_MAINNET_ADDRESS,
-    chainId: ChainId.mainnet,
-    currency: nativeCurrency,
+  // keep native assets up to date for gas price calculations
+  // NOTE: We only fetch the native asset for mainnet and chains that don't use ETH as their native token
+  const chainsToFetch = Object.entries(chainsNativeAsset).filter(
+    ([chainId, { symbol }]) => +chainId === ChainId.mainnet || symbol.toLowerCase() !== 'eth'
+  );
+  useQueries({
+    queries: chainsToFetch.map(([chainId, { address }]) => ({
+      queryKey: externalTokenQueryKey({ address, chainId: parseInt(chainId, 10), currency: nativeCurrency }),
+      queryFn: () => fetchExternalToken({ address, chainId: parseInt(chainId, 10), currency: nativeCurrency }),
+      staleTime: EXTERNAL_TOKEN_STALE_TIME,
+      cacheTime: EXTERNAL_TOKEN_CACHE_TIME,
+      enabled: !!address,
+    })),
```

Comment:
inline comment pretty much explains this, but we'll fetch eth mainnet price along with any native asset that doesn't rely on eth price